homeassistant/components/airtouch5/__init__.py

```diff
@@ -0,0 +1,46 @@
+"""The Airtouch 5 integration."""
+from __future__ import annotations
+
+from airtouch5py.airtouch5_simple_client import Airtouch5SimpleClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+PLATFORMS: list[Platform] = [Platform.CLIMATE]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Airtouch 5 from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    # Create API instance
+    host = entry.data[CONF_HOST]
+    client = Airtouch5SimpleClient(host)
+
+    # Connect to the API
+    await client.connect_and_stay_connected()
```

Comment:
What happens if the connection fails?

You should add error handling here.

Reply:
How should this be handled?
Are we meant to keep trying to connect in the background, or return a specific error?
At the moment if we fail to connect here HA reports that we failed to load and you have to reload the integration.

If we're meant to keep trying to connect in the background, how should I handle that we don't know what entities we have until we have connected?
Feel free to just point me at another component that does this nicely.

I've taken a look around in other components, `azure_event_hub` will raise a `ConfigEntryNotReady` if it fails to connect during `async_setup_entry`. Is that what you are thinking?

I really appreciate your help here.

Reply:
Looks like if you throw ConfigEntryNotReady here, home assistant will handle the retry for you

https://developers.home-assistant.io/docs/integration_setup_failures/#integrations-using-async_setup_entry

Reply:
Sorry, I have forgotten to publish my pending comment last week. See my comment above (https://github.com/home-assistant/core/pull/98136#discussion_r1358326760)

Reply:
Thanks, Done.